core/camel-core-model/src/main/java/org/apache/camel/model/RouteTemplateDefinition.java

```diff
@@ -416,10 +421,10 @@ public RouteDefinition asRouteDefinition() {
         copy.setLogMask(route.getLogMask());
         copy.setMessageHistory(route.getMessageHistory());
         copy.setOutputType(route.getOutputType());
-        copy.setOutputs(route.getOutputs());
-        copy.setRoutePolicies(route.getRoutePolicies());
+        copy.setOutputs(copy(route.getOutputs()));
+        copy.setRoutePolicies(new ArrayList<>(requireNonNullElse(route.getRoutePolicies(), emptyList())));
```

Comment:
While this would be equivalent, I wonder if adding a method such as 
```
List<T> shallowCopy(List<T> list) {
    return (list != null) ? new ArrayList<>(list) : null;
}
```
same for maps...